emar.py

```diff
@@ -62,10 +62,11 @@ def run():
 
   if DEBUG:
     print('Invoking ' + str(newargs))
-  subprocess.call(newargs, stdin=sys.stdin)
-  for d in to_delete:
-    shared.try_delete(d)
+  try:
+    return subprocess.call(newargs, stdin=sys.stdin)
+  finally:
```

Comment:
What exceptions are you trying to handle here?  I guess if the executable is missing this would throw here?   I generally don't like catch all blocks like this, but I guess for tmpfile cleanup its at least debatable.

Return code stuff lgtm

Reply:
Unless I'm misreading this, I think it just has a `finally` (for the temp files), and nothing is caught?

Reply:
Yes, but it won't handle subprocess failures because CalledProcessError because `.call` is used an not `.check_call`.   So I'm not sure it worth adding a try/catch here since we don't expect anything to be thrown.

Reply:
force of habit/overly defensive coding.  if you prefer a diff style, i don't mind.

Reply:
We already handle subprocess failures by returning non-zero at the end of this file.  So I think we just don't need this try catch in addition to that.

There are some weird I/O failures that this try might still catch, but in those cases I'm not sure we cant that much about leaving tmp files around.. they are very unexpected..

lgtm either was .. my opinions here are not super strong

Reply:
Yes, good points. Perhaps this isn't strictly needed, then. However, it does work well with the fact that we now have a return on `subprocess.call`, so the `finally` block is a nice way to do the cleanup after it.

I think we can land the PR as is (aside from adding to AUTHORS).